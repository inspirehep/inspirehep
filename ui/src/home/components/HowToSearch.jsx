import React, { useCallback, useState } from 'react';
import { Radio, Button } from 'antd';

import SpiresExamples from './SpiresExamples';
import FreetextExamples from './FreetextExamples';
import ExternalLink from '../../common/components/ExternalLink';
import { PAPER_SEARCH_URL } from '../../common/constants';

const SPIRES_RADIO = 'spires';
export const FREETEXT_RADIO = 'freetext';

function HowToSearch() {
  const [selectedRadio, setSelectedRadio] = useState(SPIRES_RADIO);
  const onRadioChange = useCallback(event => {
    setSelectedRadio(event.target.value);
  }, []);

  return (
    <div>
      <div className="mb3 tc">
        <Radio.Group value={selectedRadio} onChange={onRadioChange}>
          <Radio.Button value={SPIRES_RADIO}>SPIRES</Radio.Button>
          <Radio.Button value={FREETEXT_RADIO}>free text</Radio.Button>
        </Radio.Group>
      </div>
      <div>
        {selectedRadio === SPIRES_RADIO ? (
          <SpiresExamples />
        ) : (
          <FreetextExamples />
        )}
      </div>
      <div className="tc">
        <ExternalLink
          as={Button}
          href={PAPER_SEARCH_URL}
          type="primary"
          className="mt3"
        >
          Learn more
        </ExternalLink>
      </div>
    </div>
  );
}

export default HowToSearch;
