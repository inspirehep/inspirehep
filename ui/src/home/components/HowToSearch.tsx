import React, { useCallback, useState } from 'react';
import { Radio, Button } from 'antd';

import SpiresExamples from './SpiresExamples';
import FreetextExamples from './FreetextExamples';
import LinkWithTargetBlank from '../../common/components/LinkWithTargetBlank';
import { PAPER_SEARCH_URL } from '../../common/constants';

const SPIRES_RADIO = 'spires';
export const FREETEXT_RADIO = 'freetext';

function HowToSearch() {
  const [selectedRadio, setSelectedRadio] = useState(SPIRES_RADIO);
  const onRadioChange = useCallback(event => {
    setSelectedRadio(event.target.value);
  }, []);

  return (
    <div>
      <div className="mb3 tc">
        <Radio.Group value={selectedRadio} onChange={onRadioChange}>
          <Radio.Button value={SPIRES_RADIO}>SPIRES</Radio.Button>
          <Radio.Button value={FREETEXT_RADIO}>free text</Radio.Button>
        </Radio.Group>
      </div>
      <div>
        {selectedRadio === SPIRES_RADIO ? (
          <SpiresExamples />
        ) : (
          <FreetextExamples />
        )}
      </div>
      <div className="tc">
        <LinkWithTargetBlank
          as={Button}
          href={PAPER_SEARCH_URL}
          type="primary"
          className="mt3"
        >
          Learn more
        </LinkWithTargetBlank>
      </div>
    </div>
  );
}

export default HowToSearch;
