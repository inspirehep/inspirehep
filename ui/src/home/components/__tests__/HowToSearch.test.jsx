import React from 'react';
import { shallow } from 'enzyme';
import { Radio } from 'antd';

import HowToSearch, { FREETEXT_RADIO } from '../HowToSearch';

describe('HowToSearch', () => {
  it('renders with spires examples by default', () => {
    const wrapper = shallow(<HowToSearch />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders freetext examples after freetext radio option is selected', () => {
    const wrapper = shallow(<HowToSearch />);
    const changeEvent = { target: { value: FREETEXT_RADIO } };
    wrapper.find(Radio.Group).simulate('change', changeEvent);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
