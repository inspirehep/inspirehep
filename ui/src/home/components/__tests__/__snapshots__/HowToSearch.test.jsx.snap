// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToSearch renders freetext examples after freetext radio option is selected 1`] = `
<div>
  <div
    className="mb3 tc"
  >
    <Memo(ForwardRef)
      onChange={[Function]}
      value="freetext"
    >
      <ForwardRef(RadioButton)
        value="spires"
      >
        SPIRES
      </ForwardRef(RadioButton)>
      <ForwardRef(RadioButton)
        value="freetext"
      >
        free text
      </ForwardRef(RadioButton)>
    </Memo(ForwardRef)>
  </div>
  <div>
    <FreetextExamples />
  </div>
  <div
    className="tc"
  >
    <ExternalLink
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "Group": [Function],
          "__ANT_BUTTON": true,
          "displayName": "Button",
          "render": [Function],
        }
      }
      className="mt3"
      href="https://help.inspirehep.net/knowledge-base/inspire-paper-search"
      type="primary"
    >
      Learn more
    </ExternalLink>
  </div>
</div>
`;

exports[`HowToSearch renders with spires examples by default 1`] = `
<div>
  <div
    className="mb3 tc"
  >
    <Memo(ForwardRef)
      onChange={[Function]}
      value="spires"
    >
      <ForwardRef(RadioButton)
        value="spires"
      >
        SPIRES
      </ForwardRef(RadioButton)>
      <ForwardRef(RadioButton)
        value="freetext"
      >
        free text
      </ForwardRef(RadioButton)>
    </Memo(ForwardRef)>
  </div>
  <div>
    <SpiresExamples />
  </div>
  <div
    className="tc"
  >
    <ExternalLink
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "Group": [Function],
          "__ANT_BUTTON": true,
          "displayName": "Button",
          "render": [Function],
        }
      }
      className="mt3"
      href="https://help.inspirehep.net/knowledge-base/inspire-paper-search"
      type="primary"
    >
      Learn more
    </ExternalLink>
  </div>
</div>
`;
